### PR TITLE
Introduce landing page

### LINKED ISSUE #475: Landing Page
The current landing page at https://crystal-lang.org/reference/ is not very helpful to guide readers to the many different components of this book.

I have a proposal for a redesign, implemented as a navigation hub, pointing to many different documentation resources.

A preview is available at https://straight-shoota.github.io/crystal-book/feature/home-page
<details><summary>Screenshot</summary>

grafik

</details>

The page is structured in three sections targeted at novice users (learning materials), advanced users (specifications and manuals), and contributers (contribution materials). It also includes more links to content not hosted in the repository.

I suppose this should effectively replace https://crystal-lang.org/docs which fulfills this purpose currently, but is less extensive and actually a part of the website. I think there's great value in including the documentation landing page in this repository, because most of the contents are included here.

However, this shows another problem. Since #173, the contents of this repository are mounted at https://crystal-lang.org/reference/ - But it's become much more than just a language reference. I'd say the *Syntax and Semantics* section is the actual language specification. Maybe some other sections could also be considered part of that. But learning materials such as guides and tutorials hardly fit into that description.

The repo name `crystal-book` is actually much more open. I think we should start calling the content of this repository *Crystal Book*. The language reference is one part of it, among others.

Regarding URLs, I don't think it makes sense to mount everything under `/reference/`, like https://crystal-lang.org/reference/guides/. But that's a secondary problem, it doesn't affect user experience that much and it is considerably more complex to implement a better URL scheme. So I think we can defer that (this also depends on #384).

The important improvement is to provide a helpful landing page as a documentation hub with access to the many resources we have available.

### PR DESCRIPTION
**Pull request by straight-shoota**
Resolves #475

Preview at https://straight-shoota.github.io/crystal-book/feature/home-page/

### CODE REVIEW

**Comment by oprypin** [comment]
😍

File: docs/README.md [hunk 1]
```diff
@@ -1,15 +1,141 @@
...
+    </h3>
+    <p>Discover the ecosystem of Crystal libraries.</p>
+    <ul>
+      <li><a href="https://crystal-lang.org/community/#shards">
+        Discovering Shards
+      </a></li>
+      <li><a href="https://github.com/crystal-lang/shards/blob/master/docs/shard.yml.adoc">
+        Specification
+      </li>
+      <li><a href="the_shards_command/README.md">
```
Comment (bcardiff) [inline docs/README.md:89]: At least on the preview site this links ends in a 404. (NB: the CI is not complaining about it 🤔 )

File: docs/README.md [hunk 2]
```diff
@@ -1,15 +1,141 @@
...
+        Shards Manual
+      </a></li>
+    </ul>
+  </div>
+  <div class="card">
+    <h3><a href="using_the_compiler/README.md">
```
Comment (bcardiff) [inline docs/README.md:95]: ditto

File: docs/README.md (hunk 1 above)
Comment (straight-shoota) [reply docs/README.md:89]: Yeah, it needs `.html` extension. It seems mkdocs only recognizes (and warns) about links in markdown format and not in HTML. But I don't think we can use Markdown links inside HTML and we need HTML for the layout.

Comment (straight-shoota) [reply docs/README.md:89]: @oprypin Any ideas?

Comment (oprypin) [reply docs/README.md:89]: Sent https://github.com/straight-shoota/crystal-book/pull/1

Comment (straight-shoota) [reply docs/README.md:89]: Oh, nice 🥇

**Comment by bcardiff** [comment]
The preview site still shows invalid links. I don't know if that reflect the current status of the PR. Other than that LGTM.

**Comment by straight-shoota** [comment]
Yeah I haven't updated the preview because the last change was only structural and the visuals are the same.

**Pull request merged**